src/site/topic15.rst

```diff
@@ -3,6 +3,128 @@ Topic #15 --- ArrayLists
 ************************
 
 
+Iterator Method
+^^^^^^^^^^^^^^^
+
+.. warning::
+
+    Iterators are the focus of the next topic, so they are only presented briefly here.
+
+
+* We made use of this method in other methods by calling ``this.iterator()``
+* Which calls the method ``iterator`` from this class
+
+.. code-block:: Java
+    :linenos:
+
+    @Override
+    public Iterator<T> iterator() {
+        return new ArrayIterator<>(bag, size());
+    }
+
+* All this method does is create an instance of an ``ArrayIterator`` and return it
+* We will look at what the ``ArrayIterator`` class looks like in the next topic
+
+Add Methods
+^^^^^^^^^^^
+
+.. code-block:: Java
+    :linenos:
+
+        @Override
+        public void add(int index, T element) {
+            if (index > size()) {
+                throw new IndexOutOfBoundsException(String.format("Bag has no index %d to add to.", index));
+            }
+            if (size() == bag.length) {
+                expandCapacity();
+            }
+            shiftRight(index);
+            bag[index] = element;
+            rear++;
+        }
+
+        @Override
+        public void add(T element) {
+            add(rear, element);
+        }
+
+
+* There is nothing overly sophisticated taking place in these add methods
+    * ``add(T element)`` even hijacks ``add(int index, T element)`` for ease
```

Comment:
hijacks -> delegates to ?  
and, not only for ease, but for code/logic re-use.

Occurs again later for `remove`.